### PR TITLE
feat: create/load intervention modal

### PR DESCRIPTION
**Pull request by jlczapski**

### CODE REVIEW

**Comment by Fuhji** [comment]
cypress tests failing - but ive made quite a few changes to cypress which should resolve them in my PR

File: src/app/components/dialogs/costEffectivenessSelectionDialog/costEffectivenessSelectionDialog.component.scss [hunk 1]
```diff
@@ -1,15 +1,43 @@
+@import '../../../../assets/scss/colour';
```
Comment (Fuhji) [inline src/app/components/dialogs/costEffectivenessSelectionDialog/costEffectivenessSelectionDialog.component.scss:1]: can use `@import 'src/assets/scss/_colour.scss';` instead

**Comment by Fuhji** [comment]
image
i dont seem to get a response for ID 2?

**Comment by Fuhji** [comment]
it is possible to enter a blank name currently and submit - would want some basic verification of input

**Comment by pautva** [comment]
> Question for @bgsandan and/or @pautva - how do we stop users copy and editing an existing users copy?

I guess once we have the uuid, it should prevent users from loading the wrong intervention. But if the user loses that uuid, we would want some mechanism for them to retrieve it. 

Another solution could be to create a password for each intervention, and load them only if the passwords match.

**Review by bgsandan — COMMENTED** [review]
Just a heads up, some of the theming/styling tweaks made for tabs seems to be leaking to other bits of the app:

image

vs the develop branch (correct) version:

image

**Comment by bgsandan** [comment]
Still seeing this behaviour - re-posting in case lost in the Github issue noise.  I assume this isn't intentional?

> Just a heads up, some of the theming/styling tweaks made for tabs seems to be leaking to other bits of the app:
> 
> image
> 
> vs the develop branch (correct) version:
> 
> image

**Comment by jlczapski** [comment]
I think the issue here was that I wasn't calling 'close dialog' method in the promise callback, so it would often be closing the dialog before the data was returned and would be returning null values.. Which is probably also why it would occasionally be successful, as the data _may_ have been loaded before it was closed.

**Review by Fuhji — APPROVED** [review]

**Pull request merged**